modules/dex/options.go

```diff
@@ -0,0 +1,22 @@
+package dex
+
+import "github.com/testcontainers/testcontainers-go"
+
+// WithIssuer sets the issuer for the Dex container.
+// In most cases, the issuer should be set to the URL of the Dex container.
+// If not provided, the default issuer will be used, that will not necessarily match the actual URL of the Dex container.
+func WithIssuer(issuer string) testcontainers.ContainerCustomizer {
+	return testcontainers.CustomizeRequestOption(func(req *testcontainers.GenericContainerRequest) error {
+		req.Env["DEX_ISSUER"] = issuer
+		return nil
+	})
+}
+
+// WithLogLevel overrides the default log level for the Dex container.
+// See [log/slog](log/slog#Level) for possible values.
+func WithLogLevel(level string) testcontainers.ContainerCustomizer {
```

Comment:
question: should this take an https://pkg.go.dev/log/slog#Level if that's what it requires?

Reply:
I'd rather not to.
I thought about it and even though Dex is currently using `log/slog` it could of course change in the future.

Without any kind of compiler checking between this function and the underlying Dex version and also no guarantee about which version of Dex is used (because the image is a `string` parameter), it is perfectly possible that there's a breaking change without any testcontainers-go maintainer noticing it.
With the `logLevel` being a string parameter it's always possible to work around that and rely on the log level parsing on Dex side, whereas if we force a `log/slog.Level` here, it could become completely useless until a maintained fixes it.

But if you prefer a `log/slog.Level` I can change it of course

Reply:
thats a good call, avoid exposing the implementation detail to the user.